integration-tests/contract-permits.spec.ts

```diff
@@ -0,0 +1,242 @@
+import { CONFIGS } from "./config";
+import { MichelsonMap, MichelCodecPacker } from "@taquito/taquito";
+import { importKey } from '@taquito/signer';
+import { permit_admin_42} from "./data/permit_admin_42";
+import { permit_admin_42_expiry} from "./data/permit_admin_42_expiry";
+import { permit_admin_42_set} from "./data/permit_admin_42_set";
+
+CONFIGS().forEach(({ lib, rpc, setup, createAddress }) => {
+  const Tezos = lib;
+
+  Tezos.setPackerProvider(new MichelCodecPacker());
+
+  const FAUCET_KEY = {
+    "mnemonic": [
+      "swear",
+      "involve",
+      "mixture",
+      "pyramid",
+      "typical",
+      "swift",
+      "arch",
+      "mention",
+      "decline",
+      "mind",
+      "federal",
+      "lamp",
+      "coffee",
+      "weather",
+      "quote"
+    ],
+    // file deepcode ignore HardcodedNonCryptoSecret/test: <please specify a reason of ignoring this>
+    "secret": "782ff5c78fc58df062670fdc27d5a8a8003ecf16",
+    "amount": "32252493855",
+    "pkh": "tz1Xk7HkSwHv6dTEgR7E2WC2yFj4cyyuj2Gh",
+    "password": "zXuYCjc8aq",
+    "email": "fnctuxjz.erhkqxca@tezos.example.org"
+       }
+
+    importKey(
+      Tezos,
+      FAUCET_KEY.email,
+      FAUCET_KEY.password,
+      FAUCET_KEY.mnemonic.join(' '),
+      FAUCET_KEY.secret
+    );
+
+  describe(`Test of contracts having a permit for tzip-17: ${rpc}`, () => {
+
+    beforeEach(async (done) => {
+      await setup()
+        done()
+    })
+
+    test('Deploy a contract for admin without permits', async (done) => {
+
+      const LocalTez1 = await createAddress();
+      const localTez1Pkh = await LocalTez1.signer.publicKeyHash();
+
+      const op = await Tezos.contract.originate({
+          code: `{ parameter nat;
+            storage address;
+            code { DUP;
+                    CAR;
+                    DIP { CDR };
+                    PUSH nat 42;
+                    COMPARE;
+                    EQ;
+                    IF {  }
+                      { PUSH string "not 42";
+                        FAILWITH };
+                    DUP;
+                    SENDER;
+                    COMPARE;
+                    EQ;
+                    IF {  }
+                      { PUSH string "not admin";
+                        FAILWITH };
+                    NIL operation;
+                    PAIR }; }`,
+          storage: localTez1Pkh
+      });
+
+      await op.confirmation();
+      expect(op.hash).toBeDefined();
+      const contract = await op.contract();
+      expect(op.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
+
+      //submit the admin parameter to the contract
+      Tezos.contract
+      .at(contract.address)
+      .then((c) => {
+      return c.methods.permit("admin").send();
+      })
+      .then(async (op) => {
+        await op.confirmation(3);
+        return op.hash;
+      })
+      .catch(Error) ;
+      expect(op.hash).toBeDefined();
+      expect(op.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
```

Comment:
The lines 99 and 100 :
```
expect(op.hash).toBeDefined();      
expect(op.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY); 
```
are testing the constant `op` created at line 59 instead of the operation representing the call of the permit method

Reply:
I removed the lines.